src/library/autodj/autodjprocessor.cpp

```diff
@@ -1581,22 +1575,26 @@ void AutoDJProcessor::setTransitionMode(TransitionMode newMode) {
     }
 
     // Then re-calculate fade thresholds for the decks.
-    DeckAttributes& leftDeck = *m_decks[0];
-    DeckAttributes& rightDeck = *m_decks[1];
+    DeckAttributes* pLeftDeck = getLeftDeck();
+    DeckAttributes* pRightDeck = getRightDeck();
 
-    if (leftDeck.isPlaying() && !rightDeck.isPlaying()) {
-        calculateTransition(&leftDeck, &rightDeck, true);
-        if (rightDeck.startPos != kKeepPosition) {
+    if (!pLeftDeck || !pRightDeck) {
+        return;
+    }
```

Comment:
its inconsistent whether this case is possible (and expected under some circumstances) or an error case where a `VERIFY_OR_DEBUG_ASSERT` would be more appropriate.

Reply:
We cannot assert it, because we have controls that allow to reassign the deck to a xfader side.

Reply:
Ok, but there are other places in the code where this is asserted: https://github.com/mixxxdj/mixxx/pull/4698/files#diff-0468b47d7671b5286de23e574e3b1062d8fe118040f4cab3921ce60265eba4a6R1662-R1664

Reply:
Yes, this code is only passed when the Auto DJ is enabled and this should only be possible when we have a left and a right deck. That why I have used an assertion here in the first place. Thinking about that, we may have the case if a user changes the deck alignment during the AutoDJ run. This is not tested. 

I think I need to check that and remove the assertion as well.